### PR TITLE
Add unit test for Coriolis api v1 views

### PR DESCRIPTION
**Pull request by Cristi1324**
This PR add unit tests for Coriolis/api/v1/views/*.

### CODE REVIEW

File: coriolis/tests/api/v1/views/test_endpoint_view.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                "mapped_regions": [{'id': 'mock_id1'}, {'id': 'mock_id2'}],
+                "mock_key": "mock_value"
+            }
+
+            expected_result = {
+                'mapped_regions': ['mock_id1', 'mock_id2'],
+                'mock_key': 'mock_value'
+            }
+
+            result = endpoint_view._format_endpoint("")
```
Comment (Dany9966) [inline coriolis/tests/api/v1/views/test_endpoint_view.py:30]: Make this call with `mock.sentinel` please:
```python
endpoint = mock.sentinel.endpoint
result = endpoint_view._format_endpoint(endpoint)
# so after this call you can also have this check:
mock__format_opt.assert_called_once_with(endpoint, keys=None)
```

File: coriolis/tests/api/v1/views/test_endpoint_view.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+            self.assertEqual(
+                expected_result,
+                result
+            )
+
+    @mock.patch.object(view_utils, '_format_opt')
+    def test_format_endpoint_no_keys(self, mock__format_opt):
+            mock__format_opt.return_value = {
```
Comment (Dany9966) [inline coriolis/tests/api/v1/views/test_endpoint_view.py:39]: Please also test no mapped_regions

File: coriolis/tests/api/v1/views/test_endpoint_view.py (hunk 1 above)
Comment (Dany9966) [reply coriolis/tests/api/v1/views/test_endpoint_view.py:30]: Please do the same for all other `_format_*` tests

File: coriolis/tests/api/v1/views/test_migration_view.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            'tasks': 'mock_id1',
+            'mock_key': 'mock_value'
+        }
+
+        result = migration_view._format_migration("")
+
+        self.assertEqual(
+            mock__format_opt.return_value,
+            result
+        )
```
Comment (Dany9966) [inline coriolis/tests/api/v1/views/test_migration_view.py:74]: Add tests for `single` and `collection` please

File: coriolis/tests/api/v1/views/test_minion_pool_view.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                }
+            }],
+        }
+        mock__format_opt.return_value = mock_minion_pool_dict
+
+        result = view._format_minion_pool("")
+        self.assertEqual(
+            expected_result,
+            result
+        )
```
Comment (Dany9966) [inline coriolis/tests/api/v1/views/test_minion_pool_view.py:57]: Add tests for `single` and `collection` please

Comment (Dany9966) [reply coriolis/tests/api/v1/views/test_minion_pool_view.py:57]: Also add a test where "connection_info" and "connection_details" are empty

File: coriolis/tests/api/v1/views/test_replica_task_execution_view.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        self.assertEqual(
+            mock__format_opt.return_value,
+            result
+        )
+
+    def test_sort_tasks(self):
+        mock_tasks = [
+            {'index': 2, 'status': 'mock_status1'},
+            {'index': 3, 'status': constants.TASK_STATUS_ON_ERROR_ONLY},
+            {'index': 1, 'status': 'mock_status2'},
```
Comment (Dany9966) [inline coriolis/tests/api/v1/views/test_replica_task_execution_view.py:60]: Also pass a task without an 'index', it should default to 0

File: coriolis/tests/api/v1/views/test_replica_task_execution_view.py [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+    def test_sort_tasks_no_tasks(self):
+        expected_result = []
+
+        result = view._sort_tasks(expected_result)
+
+        self.assertEqual(
+            expected_result,
+            result
+        )
```
Comment (Dany9966) [inline coriolis/tests/api/v1/views/test_replica_task_execution_view.py:101]: Add tests for `single` and `collection` please

**Review by Dany9966 — CHANGES_REQUESTED** [review]

File: coriolis/tests/api/v1/views/test_region_view.py [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    """Test suite for the Coriolis api v1 views."""
+
+    def setUp(self):
+        super(RegionViewTestCase, self).setUp()
+        self._format_fun = region_view._format_region
+
+    @mock.patch.object(view_utils, 'format_opt')
+    def test_format_region(self, mock_format_opt):
+            mock_format_opt.return_value = {
+                "mapped_endpoints": [{'id': 'mock_id1'}, {'id': 'mock_id2'}],
```
Comment (Dany9966) [inline coriolis/tests/api/v1/views/test_region_view.py:21]: Please change `mock_id1` and `mock_id2` to `endpoint_1` / `service_1` and `endpoint_2` / `service_2` respectively, to avoid a chance these would get mixed up.

File: coriolis/tests/api/v1/views/test_minion_pool_view.py [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+        result = view._format_minion_pool(endpoint, keys)
+
+        mock_format_opt.assert_called_once_with(endpoint, keys)
+        self.assertEqual(
+            expected_result,
+            result
+        )
+
+    @mock.patch.object(view_utils, 'format_opt')
+    def test_format_minion_pool_connection_none(
```
Comment (Dany9966) [inline coriolis/tests/api/v1/views/test_minion_pool_view.py:64]: Please also add a case where there's no `minion_machines` in `minion_pool_dict`

File: coriolis/tests/api/v1/views/test_replica_task_execution_view.py [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+
+
+class ReplicaTaskExecutionViewTestCase(test_base.CoriolisApiViewsTestCase):
+    """Test suite for the Coriolis api v1 views."""
+
+    @mock.patch.object(view, '_sort_tasks')
+    @mock.patch.object(view_utils, 'format_opt')
+    def test_format_replica_tasks_execution(
+        self,
+        mock_format_opt,
```
Comment (Dany9966) [inline coriolis/tests/api/v1/views/test_replica_task_execution_view.py:19]: This mock call is not checked

File: coriolis/tests/api/v1/views/test_replica_task_execution_view.py [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+        self.assertEqual(
+            mock_format_opt.return_value,
+            result
+        )
+
+    @mock.patch.object(view, '_sort_tasks')
+    @mock.patch.object(view_utils, 'format_opt')
+    def test_format_replica_tasks_execution_no_tasks(
+        self,
+        mock_format_opt,
```
Comment (Dany9966) [inline coriolis/tests/api/v1/views/test_replica_task_execution_view.py:41]: Check call here also

**Review by Dany9966 — CHANGES_REQUESTED** [review]

File: coriolis/tests/api/v1/views/test_minion_pool_view.py [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+            result
+        )
+
+    @mock.patch.object(view_utils, 'format_opt')
+    def test_format_minion_pool_no_minion_machines(
+        self,
+        mock_format_opt,
+    ):
+        mock_minion_pool_dict = {
+            'minion_machines': [],
```
Comment (Dany9966) [inline coriolis/tests/api/v1/views/test_minion_pool_view.py:102]: This is not what I was asking for. Since the code checks whether `minion_machines` is inside the dict, you should test a case where this key is non-existent, instead of having an empty list. 
https://github.com/cloudbase/coriolis/blob/2138d7b0ae1dfdbfe44d84206c2fc5dc3c9d58b9/coriolis/api/v1/views/minion_pool_view.py#L20

**Review by Dany9966 — CHANGES_REQUESTED** [review]

**Review by Dany9966 — APPROVED** [review]
Will have to build an appliance before merging, to make sure all the API changes do not break anything. Will merge afterwards. Thank you!
